packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -57,13 +80,15 @@ export default function createVitePluginAstroServer({
 						response.end();
 						return;
 					}
-					handleRequest({
-						pipeline,
-						manifestData,
-						controller,
-						incomingRequest: request,
-						incomingResponse: response,
-						manifest,
+					localStorage.run(request, () => {
```

Comment:
Passing in the `request` which acts as the context for the local storage. When an unhandled rejection occurs, we can use this to know which request it was associated with.

Reply:
First time I asse `AsyncLocalStorage` in action!